src/hash.ts

```diff
@@ -117,16 +117,23 @@ export async function initHasher(): Promise<unknown> {
   return hasherPromise;
 }
 
-let tempHashCounter = 0;
+// Temp hashes needs to have the same length as non temp hashes. This is
```

Comment:
tricky tricky.  what goes wrong if these lengths are out of sync?

Reply:
Suggest adding a unit test that asserts that hashes from newTempHash have same length as hashes from hashOf.